protocol/communities/persistence.go

```diff
@@ -1632,3 +1632,70 @@ func (p *Persistence) SaveRequestsToJoin(requests []*RequestToJoin) (err error)
 	err = tx.Commit()
 	return err
 }
+
+func (p *Persistence) GetCuratedCommunities() (*CuratedCommunities, error) {
+	rows, err := p.db.Query("SELECT community_id, featured FROM curated_communities")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	result := &CuratedCommunities{
+		ContractCommunities:         []string{},
+		ContractFeaturedCommunities: []string{},
+	}
+	for rows.Next() {
+		var communityID string
+		var featured bool
+		if err := rows.Scan(&communityID, &featured); err != nil {
+			return nil, err
+		}
+		result.ContractCommunities = append(result.ContractCommunities, communityID)
+		if featured {
+			result.ContractFeaturedCommunities = append(result.ContractFeaturedCommunities, communityID)
+		}
+	}
+
+	return result, nil
+}
+
+func (p *Persistence) SetCuratedCommunities(communities *CuratedCommunities) error {
+	tx, err := p.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	// Clear the existing communities
+	if _, err = tx.Exec("DELETE FROM curated_communities"); err != nil {
+		return err
+	}
+
+	stmt, err := tx.Prepare("INSERT INTO curated_communities (community_id, featured) VALUES (?, ?)")
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	featuredMap := make(map[string]bool)
+	for _, community := range communities.ContractFeaturedCommunities {
+		featuredMap[community] = true
+	}
+
+	for _, community := range communities.ContractCommunities {
+		_, err := stmt.Exec(community, featuredMap[community])
```

Comment:
if we save all curated communities as featured, can it be just `_, err := stmt.Exec(community, true)` ?

Reply:
We don't save all curated communities as featured, see how `featuredMap` is propagated, am I missing something :thinking: ?

Reply:
Ah, i see, it's `communities.ContractFeaturedCommunities`